uqcsbot/dominos_coupons.py

```diff
@@ -226,69 +249,64 @@ def _get_coupons_from_page(source: str) -> List[Coupon]:
-
-            if (
-                not expiry_date_container
-                or not description_container
-                or not code_container
-            ):
-                continue
-
-            expiry_date_str: str = expiry_date_container.get_text(strip=True)
-            description: str = description_container.get_text(strip=True)
-            code: str = code_container.get_text(strip=True)
-
-            temp_code: str = code.replace(",", "")
-            if not temp_code.isdigit():
-                continue
-
-            code = code.replace(",", " ")
-
-            if url == FRUGAL_FEEDS_DOMINOS_URL:
-                date_values: List[str] = expiry_date_str.split()
-                try:
-                    # Convert shortened month to numerical value
-                    month: int = datetime.strptime(date_values[1], "%b").month
-                    expiry_date_str = "{year}-{month}-{day}".format(
-                        year=int(date_values[2]), month=month, day=int(date_values[0])
-                    )
-                except (ValueError, IndexError):
-                    pass
-
-            coupon = Coupon(code, expiry_date_str, description)
-            coupons.append(coupon)
+    coupons: List[Coupon] = []
+    url = urls[source]
+
+    http_response: requests.Response = requests.get(url)
+    if http_response.status_code != requests.codes.ok:
+        raise HTTPResponseException(http_response.status_code, url)
+
+    soup = BeautifulSoup(http_response.content, "html.parser")
+    soup_coupons: List[BeautifulSoup] = []
+
+    if url == COUPONESE_DOMINOS_URL:
+        soup_coupons = soup.find_all(class_="ov-coupon")
+    elif url == FRUGAL_FEEDS_DOMINOS_URL:
+        tables = soup.select('[class^="tablepress"]')
+        for table in tables:
+            # Headers have stuff we don't want
+            rows = table.find_all("tr")[1:]
+            soup_coupons.extend(rows)
+
+    siteclass: Dict[str, str] = website_coupon_classes.get(url, {})
+
+    for soup_coupon in soup_coupons:
+        expiry_date_container = soup_coupon.find(class_=siteclass.get("expiry"))
+        description_container = soup_coupon.find(class_=siteclass.get("description"))
+        code_container = soup_coupon.find(class_=siteclass.get("code"))
+
+        if not expiry_date_container or not description_container or not code_container:
+            continue
+
+        expiry_date_str: str = expiry_date_container.get_text(strip=True)
+        description: str = description_container.get_text(strip=True)
+        code: str = code_container.get_text(strip=True)
+
+        # Check if code is a digit - take seperators out so we can check
+        # this properly
+        temp_code: str = code.replace(",", "").replace(" ", "")
+        if not temp_code.isdigit():
+            continue
+
+        # Keep formatting same for coupons with multiple codes
+        if source == "frugalfeeds":
+            code = code.replace(",", ", ")
+        else:
```

Comment:
Instead of using `else`, it might be best to use `if source == "couponese"`, as this could be forgotten if we add more sources in the future.